ambry-replication/src/test/java/com.github.ambry.replication/MockHost.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.replication;
+
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.store.StoreKeyConverter;
+import com.github.ambry.utils.SystemTime;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+
+/**
+ * Representation of a host. Contains all the data for all partitions.
+ */
+public class MockHost {
```

Comment:
InMemoryHost?

Reply:
I struggled here. This is not true in memory, it just a wrapper of `InMemoryStores`.
AND when `InMemory` is used, it usually related to storage, but host here is not.

Reply:
A similar case is `MockConnectionPool`.